src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+	}
+
+	return onlyServiceNames
+}
+
+templ MappingRegularForm(vhostsWithMappings []dto.VirtualHostWithMappings) {
+	<form
+		hx-post="/api/v1/vhosts/mapping/"
+		hx-indicator="#loading-overlay"
+		hx-swap="none"
+	>
+		<div class="-mt-4 my-6 space-y-6">
+			@componentMappings.MappingTypeSelectInput("targetType", "Type", "mapping.targetType", "clearMappingTargetValue()")
+			@componentMappings.MappingTypeInfoDialog()
+			<template x-if="isCreateMappingFromVhost">
+				@componentForm.ReadOnlyField("text", "hostname", "Hostname", "virtualHost.hostname", false)
+			</template>
+			<template x-if="!isCreateMappingFromVhost">
+				@componentForm.SelectInput("hostname", "Hostname", "virtualHost.hostname", filterOnlyVhostsHostnames(vhostsWithMappings), true, false)
+			</template>
+			@componentForm.InputFieldWithPrefix("text", "path", "Source URL", "mapping.path", "vhostWithTrailingSlash", false)
+		</div>
+		<template x-if="mapping.targetType == 'service'">
+			<div class="mb-6 space-y-6">
+				@componentForm.SelectInput("targetValue", "Service Name", "mapping.targetValue", getOnlyServiceNames(), true, false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'url'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "targetValue", "Destination URL", "mapping.targetValue", false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'inline-html'">
+			<div class="mb-6 space-y-6">
+				@componentForm.TextArea("targetValue", "Inline HTML", "mapping.targetValue", false)
+				@componentMappings.ResponseCodeSelectInput("targetHttpResponseCode", "Response Code", "mapping.targetHttpResponseCode")
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'response-code'">
+			<div class="mb-6 space-y-6">
+				@componentMappings.ResponseCodeSelectInput("targetValue", "Response Code", "mapping.targetValue")
+				<input type="hidden" x-model="mapping.targetValue" x-bind:value="mapping.targetHttpResponseCode"/>
+			</div>
+		</template>
+		<section class="mb-6 w-full rounded-md">
+			<details class="bg-os-500 group rounded-md p-4 group-open:bg-opacity-30" x-bin:open="isAdvancedSettingsClosed">
+				<summary class="relative flex cursor-pointer list-none items-center pr-8 text-neutral-50 text-opacity-80 focus-visible:outline-none group-open:text-opacity-100 group-hover:text-opacity-100 [&::-webkit-details-marker]:hidden">
+					<i class="ph-duotone ph-gear mr-2 text-2xl"></i>
+					<span class="font-bold">Advanced Settings</span>
+					<i class="ph-bold ph-plus absolute right-0 top-1.5 text-xs transition duration-300 group-open:rotate-45"></i>
+				</summary>
+				<div class="mt-5 py-2">
+					@componentMappings.MatchPatternSelectInput("matchPattern", "Match Pattern", "mapping.matchPattern")
+				</div>
+			</details>
+		</section>
+		<template x-if="virtualHost.hostname == '' || ((mapping.targetType != 'response-code' && mapping.targetType != 'static-files') && mapping.targetValue == '')">
```

Comment:
This def should be a method. Dude, I can't read this if. Argh.

Reply:
Done!